### DISCUSSION TITLE
Validate property before updating, don't update if validation fails

### DISCUSSION THREAD

**Discussion by DanielGSoftware** [livewire/livewire#2349] [1. Help]
Hey everyone, I was wondering if it's possible to validate a property before it's being updated, and if the validation fails the property should not be updated. 
I'm currently trying to accomplish this with
``` 
public function updating(string $attribute): void
    {
        $this->validateOnly($attribute);
    }
```

But even if validation fails, the property stills gets updated. This is not what I want, as demonstrated in this example below, it will give errors when `age` is set to a string by an input field, because the typed property is `int`. 

```
class Cat extends Component {

    public int $age = 0;

    public function rules(): array
    {
        return [
            'age' => ['integer']
        ];
    }

    public function updating(string $attribute): void
    {
        $this->validateOnly($attribute);
    }
}
```

Does anyone know if there's a solution for this?

**Comment by jamesRUS52** [comment]
I need answer also.
It look very strange while updateting data on modal window and it data display in main page. When validating fail in modal window the inccorect data already updated and display on main page

**Comment by jameslkingsley** [comment]
This could be a quick fix [implemented here](https://github.com/livewire/livewire/blob/master/src/ComponentConcerns/HandlesActions.php#L78).

Something like:
```php
if ($this->updating($name, $value) === false) {
    return;
}
```

And then you'd have the option to completely bail on the update by doing this:
```php
public function updating($attribute, $value)
{
    if (/* should not continue */) {
        return false;
    }
}
```

**Reply by stefanocurnis** [reply]
Do you ever solve this? I'm having the same problem when binding to model property which is a Carbon date. While I'm typing the date, it throws the error that Carbon can't parse.
